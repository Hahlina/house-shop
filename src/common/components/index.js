export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
