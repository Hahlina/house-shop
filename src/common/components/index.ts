export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
