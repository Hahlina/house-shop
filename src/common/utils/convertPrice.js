export const convertPrice = (price) =>
    new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: 'USD'
    }).format(price || 0);
