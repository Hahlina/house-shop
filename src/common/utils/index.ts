export * from './convertPrice';
export * from './capitalizeFirstLetter';
