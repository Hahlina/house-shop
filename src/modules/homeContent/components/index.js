export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
