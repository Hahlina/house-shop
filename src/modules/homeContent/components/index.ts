export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
