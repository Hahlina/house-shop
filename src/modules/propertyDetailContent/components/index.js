export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
