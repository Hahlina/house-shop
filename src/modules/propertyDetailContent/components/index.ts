export * from './PropertyGallery/PropertyGallery';
export * from './PropertyGallery/PropertyGalleryLoader';
export * from './FullInformationCard/FullInformationCard';
export * from './FullInformationCard/FullInformationCardLoader';
export * from './PropertyDetailContentLoader/PropertyDetailContentLoader';
export * from './PropertyDetailBreadCrumbs/PropertyDetailBreadCrumbs';
