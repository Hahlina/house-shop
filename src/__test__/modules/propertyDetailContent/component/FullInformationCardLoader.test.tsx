import { FullInformationCardLoader } from 'modules/propertyDetailContent/components';

import { render, screen } from 'common/utils/test';

describe('FullInformationCardLoader', () => {
    it('should renders the correct number of Skeleton components', () => {
        render(<FullInformationCardLoader />);
        expect(screen.getAllByTestId('skeleton')).toHaveLength(6);
    });

    it('should renders the correct height of Skeleton components', () => {
        render(<FullInformationCardLoader />);
        const skeletonComponents = screen.getAllByTestId('skeleton');
        skeletonComponents.forEach((skeleton) => {
            expect(skeleton).toHaveStyle('height: 30px');
        });
    });

    it('should take a snapshot', () => {
        const { container } = render(<FullInformationCardLoader />);
        expect(container).toMatchSnapshot();
    });
});
