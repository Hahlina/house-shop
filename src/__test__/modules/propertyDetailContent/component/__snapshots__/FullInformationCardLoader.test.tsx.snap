// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullInformationCardLoader should take a snapshot 1`] = `
<div>
  <div
    class="sc-csKJxZ imMYB"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
      data-testid="skeleton"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
      data-testid="skeleton"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
      data-testid="skeleton"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
      data-testid="skeleton"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
      data-testid="skeleton"
    />
  </div>
</div>
`;
